### DISCUSSION TITLE
[Feature] allow disable email and password auth completely

### DISCUSSION THREAD

**Discussion by winston0410** [johanohly/AirTrail#87] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I would like to see a new env that allows me to completely disable the email and password auth, and for my user to login through OIDC. I would also like to see a button prompting user to login with my OIDC provider

**Comment by johanohly** [comment]
Thank you for your feature request! 
Enabling "Auto Login" should theoretically prevent your users from logging in through email and password, unless your use-case is added security? 

Regarding your request to see a button for OIDC login, I am not sure I follow. There is a button to log in via OIDC on the login page already.
![image](https://github.com/user-attachments/assets/31792ad2-60b1-4ec7-8940-dcc68b308f0e)

**Reply by winston0410** [reply]
I would like to see those username and password field to be hidden from the UI, because even if I have Auto Login enabled, it will still takes time for the OAuth2 flow to happen. User might attempt to complete the form anyway, during the redirect for the OIDC flow.

**Reply by johanohly** [reply]
I can see why one might want to hide the email and password form all together rather than just getting redirected. 
Although all buttons *are* disabled while any loading is happening, including during the OAuth flow, so I think the question becomes whether to hide the form when "Auto Login" is enabled, rather than whether to add a whole extra option for it.

**Reply by winston0410** [reply]
Yes I agree, that is a good interpretation. Do you think you would accept an env, say `HIDE_PASSWORD_LOGIN_FORM`, so this can be achieved without building any UI for it? I can create a PR for that if you want

**Reply by johanohly** [reply]
To be honest I am not sure this should be a separate config value. 

I think the form should be hidden when "Auto Login" is enabled instead of showing it. I can't think of a use-case where you would want the form showing while the user is getting automatically redirected to OAuth.

**Reply by winston0410** [reply]
I am happy with both, and I can't think of any use-case to show the form too, when `Auto-login` is enabled.

**Comment by johanohly** [comment]
This will be released in the next release coming up hopefully pretty soon!

**Comment by johanohly** [comment]
Released in v1.0.0!

**Comment by adamzvolanek** [comment]
Would it be possible to disable login entirely?
